Java: Document changelog deletion (#1969)
cap-js/docs: java/change-tracking.md

- The data type of the changed attribute.
- The technical path from the root entity to the tracked target entity.

By default, changes remain in your database even when their entities are deleted.

> If you want to delete changelogs together with your entities, your **domain entity** must be extended with `changelog.changeTracked` aspect.

Reviewer: ```suggestion
If you want to delete changelogs together with your entities, your **domain entity** must be extended with the `changelog.changeTracked` aspect.
```

> To enable deletion of changes per entity, annotate domain entity like this:

Reviewer: ```suggestion
To enable the deletion of changes per entity, annotate your domain entity like this:
```

```cds

> annotate model.Books.changes:change with @cascade: { delete };

Reviewer 2: Why can't we annotate `model.Books.changes with @cascade: { delete };` ? What's the difference to annotating `model.Books.changes:change with @cascade: { delete };`?

```

> To enable this for all entities in the model, add the following annotation to your model:

Reviewer: ```suggestion
To enable deletion of changes for all entities in the model, add the following annotation to your model:
```
